libraries/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -265,6 +310,90 @@ export class PackageJsonUpdater {
     }
   }
 
+  /**
+   * Remove a dependency from a particular project. The core business logic for "rush remove".
+   */
+  public async doRushRemoveAsync(options: IPackageJsonUpdaterRushRemoveOptions): Promise<void> {
+    const { projects, packagesToRemove, skipUpdate, debugInstall } = options;
+
+    console.log();
+    const dependenciesToRemove: Record<string, DependencyType[]> = {};
+
+    const allPackageUpdates: IRemoveProjectOptions[] = [];
+
+    for (const project of projects) {
+      for (const { packageName } of packagesToRemove) {
+        const dependencyTypes: DependencyType[] = [];
+        const dependency: PackageJsonDependency | undefined =
+          project.packageJsonEditor.tryGetDependency(packageName);
+        const devDependency: PackageJsonDependency | undefined =
+          project.packageJsonEditor.tryGetDevDependency(packageName);
+        const resolution: PackageJsonDependency | undefined =
+          project.packageJsonEditor.tryGetResolution(packageName);
+        if (dependency) {
+          dependencyTypes.push(dependency.dependencyType);
+        }
+        if (devDependency) {
+          dependencyTypes.push(devDependency.dependencyType);
+        }
+        if (resolution) {
+          dependencyTypes.push(resolution.dependencyType);
+        }
+        if (!dependencyTypes.length) {
+          throw new Error(`The project "${project.packageName}" do not have ${packageName} in package.json.`);
```

Comment:
Seems like we should probably just print an informational message here instead of a hard error. The user wants the dependency removed, so a state where it's not in the dependency list at the end of the run is what's expected even if it wasn't there in the first place.

Reply:
done